sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -318,8 +318,20 @@ public EncryptedBlobClientBuilder credential(TokenCredential credential) {
      * @throws NullPointerException If {@code sasToken} is {@code null}.
      */
     public EncryptedBlobClientBuilder sasToken(String sasToken) {
-        this.sasTokenCredential = new SasTokenCredential(Objects.requireNonNull(sasToken,
-            "'sasToken' cannot be null."));
+        return this.sasToken(new AzureSasCredential(Objects.requireNonNull(sasToken,
+            "'sasToken' cannot be null.")));
+    }
+
+    /**
+     * Sets the SAS token used to authorize requests sent to the service.
+     *
+     * @param credential The SAS token to use for authenticating requests.
+     * @return the updated EncryptedBlobClientBuilder
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    public EncryptedBlobClientBuilder sasToken(AzureSasCredential credential) {
```

Comment:
Should this method be named `azureSasCredential` or `sasCredential` instead as it is taking the credential type and not the token?

Reply:
https://github.com/Azure/azure-sdk-for-java/pull/18473#discussion_r552982188 .
I was thinking about positioning as an overload to `sasToken`. end of day they do same thing just take different type as input.
I'm fine either way.

Reply:
I think the distinction is that a `credential` allows you to update the token while a `token` cannot be updated at runtime. I am leaning more towards calling it `sasCredential()` but don't have a strong opinion either way.

Reply:
Yeah, the prefix should group them anyway. I'll make it `sasCredential` then.

Reply:
Maybe this should be just called `credential`?

![image](https://user-images.githubusercontent.com/61715331/104066386-2e713180-51b6-11eb-872a-401e0b40eb8d.png)

Reply:
I would lean towards `credential` as the guideline also seems to lean that direction, in my opinion, https://azure.github.io/azure-sdk/java_introduction.html#java-service-client-builder-consistency.

Reply:
renamed to `credential()` to align with guideline and rest of builders that take credential.